securetea/users.py

```diff
@@ -0,0 +1,92 @@
+# -*- coding: utf-8 -*-
+u"""User Logging module for SecureTea.
+
+Project:
+    ╔═╗┌─┐┌─┐┬ ┬┬─┐┌─┐╔╦╗┌─┐┌─┐
+    ╚═╗├┤ │  │ │├┬┘├┤  ║ ├┤ ├─┤
+    ╚═╝└─┘└─┘└─┘┴└─└─┘ ╩ └─┘┴ ┴
+
+    Author: Kushal majmundar <majmundarkushal9@gmail.com> , April 16 2019
+    Version: 1.0
+    Module: SecureTea
+
+"""
+import time
+import sqlite3
+import psutil
+
+connection = sqlite3.connect('/etc/securetea/db.sqlite3')
+
+connection.execute('''CREATE TABLE IF NOT EXISTS USERS(
+    ID INTEGER PRIMARY KEY AUTOINCREMENT,
+    NAME CHAR(100),
+    IP CHAR(100),
+    dt TIMESTAMP
+    );''')
+
+class SecureTeaUserLogger():
+    """Initilize the logger for the script.
+
+    Attributes:
+        BLUE (str): Blue color
+        BOLD (str): Bold String
+        ENDC (str): End Char
+        ERROR (TYPE): Red color
+        LEGEND (TYPE): Legend with module name and Datetime
+        OKGREEN (TYPE): Green color
+        VIOLET (str): Violet  color
+        WARNING (TYPE): Orange color
+        YELLOW (str): Yellow color
+
+    """
+
+    BOLD = '\033[1m'
+    ENDC = '\033[0m'
+    BLUE = '\033[94m'
+    VIOLET = '\033[95m'
+    OKGREEN = '\033[92m' + BOLD + "Info : " + ENDC + '\033[92m'
+    WARNING = '\033[93m' + BOLD + "Warn : " + ENDC + '\033[93m'
+    ERROR = '\033[91m' + BOLD + "Error: " + ENDC + '\033[91m'
+    YELLOW = '\033[33m'
+
+    def __init__(self, debug=False):
+        """Init logger params.
+
+        Args:
+            debug (bool): Script validity
+        """
+
+        self.LEGEND = self.VIOLET + '[' + ']' + \
+            '  ' + self.YELLOW + '[ ' + \
+            str(time.strftime("%Y-%m-%d %H:%M")) + ' ]  '
+        self.debug = debug
+
+    def addUsers(self):
+        """
+        add user info. and return new users added
+        """
+        message = "USERS UPDATES\n"
+        cur_users = []
+        for user in psutil.users():
+            cur_users.append((user.name, user.host))
+        # print(cur_users)
+        users = list(connection.execute("SELECT NAME,IP FROM USERS"))
+        for user in users:
```

Comment:
Please check if the `users` list is not None, else this can give a `TypeError`

Reply:
Also, don't you think we must check for new users and add them to the DB before checking which users have logged out? Why this approach? This will possibly eliminate DB returning None, **addition before deletion.**

Reply:
Also, please account for this case:
1. If the user didn't log out before our program terminated then our DB still has record for that user. Next time when the program starts and the user SSHes into the server, message for new user login will not be notified? Don't you think a smart session based approach based on timestamp can be used to eliminate this problem?

Reply:
+ DB returning None is not a problem. Any order could be followed
+ The goal is that whenever the script it running the system should be secure. Hence if the user logs in later on notifying is not a task. Consider the following scenario that a user has been continuously logged in during the time the script has been stopped. Hence according to you, a notification should be sent that he is still logged in which makes no sense. We do not need to take the scenario you have mentioned into account. We only need to notify that activities when the script is running as it may result in a breach of privacy.

Reply:
@kUSHAL0601 What if the DB returns None, then don't you think typecasting None i.e. `list(None)` can give an error?